app/views/spree/admin/products/new.html.haml

```diff
@@ -11,7 +11,7 @@
           = f.field_container :supplier do
             = f.label :supplier_id, t(".supplier")
             %span.required *
-            = f.collection_select(:supplier_id, @producers, :id, :name, {:include_blank => true}, {:class => "select2 fullwidth"})
+            = f.collection_select(:supplier_id, @producers, :id, :name, {:include_blank => true, :selected => select_only_item(@producers.size)}, {:class => "select2 fullwidth"})
```

Comment:
Yes we can also do that. @RachL decscribed the solution I implemented in the issue. Please let me know which option is more appropriate and I can update my PR

Reply:
@andrewpbrett I'm not sure it makes a lot of sense? It can lead to errors? The point here was to make it easier for single producer profiles. Big hubs will always need to select a producer, so best to keep it has a mandatory field.

Reply:
Thanks for clarifying @RachL and @sinansonmez, let's go with that: the field should be empty if there is more than one option, so that they are force to choose; but if there is exactly one option, it should be prepopulated.

Reply:
Thanks, I just updated the PR accordingly. Please also see the screen recording below

https://user-images.githubusercontent.com/37421564/136091345-5a140626-ef6d-48da-86e4-bd0c635d8bd5.mov